server/src/main/java/org/elasticsearch/action/ingest/GetPipelineResponse.java

```diff
@@ -135,6 +135,10 @@ public boolean equals(Object other) {
                     if (pipeline.equals(otherPipeline) == false) {
                         return false;
                     }
+                    otherPipelineMap.remove(pipeline.getId());
+                }
+                if (otherPipelineMap.isEmpty() == false) {
+                    return false;
```

Comment:
This change will be reverted from this PR, and addressed separately through #93695